doc/api/http2.md

```diff
@@ -0,0 +1,692 @@
+creates and returns an `Http2Stream` instance that can be used to send an
+HTTP/2 request to the connected server.
+
+This method is only available if `http2session.type` is equal to
+`http2.constants.NGHTTP2_SESSION_CLIENT`.
+
+(TODO: fill in detail)
+
+#### http2session.rstStream(stream, code)
+
+* stream {Http2Stream}
+* code {number}
+
+Sends an `RST-STREAM` frame to the connected HTTP/2 peer, causing the given
+`Http2Stream` to be closed on both sides using error code `code`.
+
+#### http2session.setTimeout(msecs, callback)
+
+* `msecs` {number}
+* `callback` {Function}
+
+(TODO: fill in detail)
+
+#### http2session.shutdown(options[, callback])
+
+* `options` {Object}
+  * `graceful` {boolean} `true` to attempt a polite shutdown of the
+    `Http2Session`.
+  * `immediate` (boolean) `true` to force the shutdown to occur immediately,
+    regardless of any data transfer that may be pending.
+  * `errorCode` {number} The HTTP/2 Error Code to return. Note that this is
+    *not* the same thing as an HTTP Response Status Code.
+  * `lastStreamID` {number} The Stream ID of the last successfully processed
+    `Http2Stream` on this `Http2Session`.
+  * `opaqueData` {Buffer} A `Buffer` instance containing arbitrary additional
+    data to send to the peer upon disconnection. This is used, typically, to
+    provide additional data for debugging failures, if necessary.
+* `callback` {Function}
+
+Attempts to shutdown this `Http2Session` using HTTP/2 defined procedures.
+If specified, the given `callback` function will be invoked once the shutdown
+process has completed.
+
+#### http2session.socket
+
+* Value: {net.Socket|tls.TLSSocket}
+
+A reference to the [`net.Socket`][] or [`tls.TLSSocket`][] to which this
+`Http2Session` instance is bound.
+
+#### http2session.state
+
+* Value: {Object}
+
+An object describing the current status of this `Http2Session`.
+(TODO: fill in detail)
+
+#### http2session.submitPriority(stream, options)
+
+* `stream` {Http2Stream}
+* `options` {Object}
+  * `exclusive` {boolean} (TODO: fill in detail)
+  * `parent` {number} (TODO: fill in detail)
+  * `weight` {number} (TODO: fill in detail)
+
+Updates the priority for the given `Http2Stream` instance. If `options.silent`
+is `false`, causes a new `PRIORITY` frame to be sent to the connected HTTP/2
+peer.
+
+#### http2session.submitSettings(settings)
+
+* `settings` {[Settings Object][]}
+* Returns {undefined}
+
+Updates the current local settings for this `Http2Session` and sends a new
+`SETTINGS` frame to the connected HTTP/2 peer.
+
+#### http2session.type
+
+* Value: {number}
+
+The `http2session.type` will be equal to
+`http2.constants.NGHTTP2_SESSION_SERVER` if this `Http2Session` instance is a
+server, and `http2.constants.NGHTTP2_SESSION_CLIENT` if the instance is a
+client.
+
+### Class: Http2Stream
+
+* Extends: {Duplex}
+
+Instances of the `Http2Stream` class represent a bidirectional HTTP/2
```

Comment:
Making 'Instances' singular (e.g. 'Each instance') might improve the readability of this sentence. Also, would it be within the scope of this section to give a brief overview of when streams are created? For example, to mention that on the client, it's returned by `Http2Session::request`, while on the server, it's passed through the `'stream'` event.

In addition, nit in the last sentence -- 'its lifetime'.